src/gui/editors/PianoRoll.cpp

```diff
@@ -628,47 +636,112 @@ void PianoRoll::markSemiTone(int i, bool fromMenu)
 }
 
 
+
+
+/*! \brief Copy notes to the ghost note vector
+ *
+ *  This is a private method. It will also update GUI components related to ghost notes.
+ */
+void PianoRoll::setGhostNotes(const NoteVector& notes)
+{
+	// TODO undo ghost note changes
+
+	m_ghostNotes.clear();
+	for (auto& note: notes)
+	{
+		m_ghostNotes.push_back(new Note(note->length(), note->pos(), note->key()));
+	}
+	if (auto bounds = boundsForNotes(m_ghostNotes))
+	{
+		m_ghostBounds = bounds.value();
```

Comment:
It's safe to use `*bounds` here since you already checked for a value using `if`.